api/index.php

```diff
@@ -29,56 +30,107 @@
 
 session_cache_limiter(false);
 
-$app = new Slim(array(
-    'mode' => $mode == 'production' ? 'production' : 'development'
-));
+$app = setupApplication($app, $mode);
 
-$app->configureMode('production', function () use ($app) {
-    $app->config(array(
-        'log.enable' => true,
-        'debug' => false
-    ));
-});
-
-$app->configureMode('development', function () use ($app) {
-    $app->config(array(
-        'log.enable' => false,
-        'debug' => true
-    ));
-});
+register_shutdown_function('session_write_close'); // prevents unexpected effects when using objects as save handlers
 
-$app->get('/options', function () use ($app) {
-    global $motd, $authentication_type, $cas_url, $cas_sso, $package_description, $facility_courier_countries, $facility_courier_countries_nde, $dhl_enable, $dhl_link, $scale_grid, $preset_proposal, $timezone, $valid_components, $enabled_container_types;
-    $app->contentType('application/json');
-    $app->response()->body(json_encode(array('motd' => $motd, 'authentication_type' => $authentication_type, 'cas_url' => $cas_url, 'cas_sso' => $cas_sso, 'package_description' => $package_description, 'facility_courier_countries' => $facility_courier_countries, 'facility_courier_countries_nde' => $facility_courier_countries_nde, 'dhl_enable' => $dhl_enable, 'dhl_link' => $dhl_link, 'scale_grid' => $scale_grid, 'preset_proposal' => $preset_proposal, 'timezone' => $timezone, 'valid_components' => $valid_components, 'enabled_container_types' => $enabled_container_types)));
-});
+setupDependencyInjectionContainer($app, $isb, $port);
 
-// the following prevents unexpected effects when using objects as save handlers
-register_shutdown_function('session_write_close');
+$auth = $app->container['auth'];
+$auth->validateAuthentication();
+$auth->updateActivityTimestamp();
 
-$port = array_key_exists('port', $isb) ? $isb['port'] : null;
+$app->container['dispatch']->dispatch();
 
-// MySQL database class hard-coded
-$db = new MySQL($isb['user'], $isb['pass'], $isb['db'], $port);
-
-// Alternatively, use dynamic class instantiation.
-// Database type ($dbtype) specified in config.php.
-// $db = Database::get();
-
-$db->set_app($app);
```

Comment:
This is missing from the new code and is needed because a content header is set depending on it being set. We should add the app to the constructor.

Reply:
It isn't, and it is hard to add it to the constructor so I have moved it into the DI code.